scripts/Muon/GUI/Common/plot_widget/fit_pane/plot_fit_pane_presenter.py

```diff
@@ -10,8 +10,9 @@
 
 class PlotFitPanePresenter(BasePanePresenter):
 
-    def __init__(self, view, model, context,figure_presenter):
-        super().__init__(view, model, context,figure_presenter)
+    def __init__(self, view, model, context, fitting_context, figure_presenter):
```

Comment:
We shouldn't pass the fitting context in here, it just clutters up the arguments for no benefit. We can get the methods via longhand (context.fitting_context) or by having `self.fitting_context = context.fitting_context` in the init.

Reply:
The fitting context needs to be passed in at this level because different panes will have a different `FittingContext`. For instance the Fitting tab passes in `self.context.fitting_context`, while the Model fitting tab passes in `self.context.model_fitting_context`

Reply:
we can place any calls to the fitting context in their own functions, then use inheritance to override them with custom presenters. I think that might be cleaner

Reply:
can we have a comment to clarify that the `fitting_context` is the specific one corresponding to the plot (e.g. model fitting is the model fitting context)?